### ISSUE TITLE
How to classify a sample as unknown?

### ISSUE THREAD

**Issue by FunnyDragonK** [sgvaze/osr_closed_set_all_you_need#5]
Hello! I am glad to see this work when I encounter OSR problems in practice.
I'm wonder that how to give the unknown label to a sample with the classifier trained on the closed set.

**Comment by sgvaze** [comment]
Hi, I'm glad this problem is of practical interest! 

For a K-way classification task, the cross-entropy baseline outputs a K-dimensional vector. The maximum value of this vector (with or without Softmax normalisation) gives a 'score' that the image belongs to _any_ of the K classes. To label an image as 'seen' of 'unseen', a threshold must be chosen for this score. However, the evaluation metric in this paper is 'AUROC', which aggregates 'seen'/'unseen' classification performance over all choices of threshold (i.e the metric is threshold agnostic). An example evaluation can be seen [here](https://github.com/sgvaze/osr_closed_set_all_you_need/blob/5b5a37cd5c230f5c4b1fd3106e46a63e4f18b933/methods/tests/openset_test.py#L53).

In practise, a threshold could be chosen by looking at 'seen'/'unseen' classification performance on a validation set of open-set images. The best choice of threshold would depend on the application (e.g are False Positives or False Negatives more important).

**Comment by FunnyDragonK** [comment]
Got it. 
Thanks for the detailed answer!
